### PR TITLE
pdms: remove shutdown component and add test for scaler&upgrader

### LINKED ISSUE #5553: (pdms)modify tso config did not take effect
## Bug Report

**What version of Kubernetes are you using?**

**What version of TiDB Operator are you using?**

**What storage classes exist in the Kubernetes cluster and what are used for PD/TiKV pods?**

**What's the status of the TiDB cluster pods?**

**What did you do?**

add config tso-save-interval = "10s" according edit tc like following:
```
  pdms:
  - baseImage: hub-new.pingcap.net/qa/pd
    config: |
      tso-save-interval = "10s"
      tso-update-physical-interval = "1ms"
      [log]
        [log.file]
          filename = "/var/lib/pd/tso/tso.log"
    limits:
      cpu: "1"
      memory: 2Gi
      storage: 10Gi
    name: tso
```

**What did you expect to see?**
Works.

**What did you see instead?**
didn't take affect.

### PR DESCRIPTION
**Pull request by HuSharp**
### What problem does this PR solve?

Fix the bug that modifing tso config did not take effect

Closes #5553, related https://github.com/tikv/pd/issues/7519

### Tests

- [ ] Unit test 
- [ ] Manual test 

### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note

```

### CODE REVIEW

**Comment by HuSharp** [comment]
@csuzhangxc friendly ping :)

File: pkg/manager/member/pd_ms_member_manager.go [hunk 1]
```diff
@@ -64,10 +64,35 @@ func (m *pdMSMemberManager) Sync(tc *v1alpha1.TidbCluster) error {
...
+
+			oldPDMSSet := oldPDMSSetTmp.DeepCopy()
+			newPDMSSet := oldPDMSSetTmp.DeepCopy()
+			*newPDMSSet.Spec.Replicas = 0
+			if err := m.scaler.Scale(tc, oldPDMSSet, newPDMSSet); err != nil {
+				return err
+			}
+			mngerutils.UpdateStatefulSetWithPrecheck(m.deps, tc, "FailedUpdatePDMSSTS", newPDMSSet, oldPDMSSet)
+		}
+		klog.Infof("PD Micro Service is enabled, but PD is not enabled or not in `ms` mode, skip syncing PD Micro Service")
```
Comment (csuzhangxc) [inline pkg/manager/member/pd_ms_member_manager.go:92]: it seems this log message is not accurate.

**Review by csuzhangxc — COMMENTED** [review]
rest LGTM

**Review by csuzhangxc — APPROVED** [review]

**Pull request merged**